src/frontend/middleware.go

```diff
@@ -200,9 +201,10 @@ func (tw traceware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	start := time.Now()
 	defer func() {
+		labels = append(labels, semconv.HTTPAttributesFromHTTPStatusCode(rr.status)...)
```

Comment:
Any idea why this isn't being added in line 200?

Reply:
That code needs the response that isn't available until after line 207.

Reply:
The "defer" keyword indicates that the function will be ran after the scope it is in has finished. The http handler won't write the status code until after ServeHTTP is finished, so the code to instrument the status code has to be written in a defer function (or after the call to ServeHTTP, but this pattern was present already).

Reply:
What in the OTEL spec do you consider "manual instrumentation" vs. I guess automatic instrumentation like we would see in an agent? Everything I've been getting on my end is that if a customer is sending us OTEL data, we should be finding ways to display it.

Reply:
There are (roughly) 3 types of instrumentation:
- Automatic instrumentation: the presence of some library automatically injects itself and adds instrumentations. E.g. otel java agent.
- Library instrumentation: a library that you can add to your library, or that wraps your library and add instrumentation. E.g. [these grpc interceptors](https://github.com/newrelic-forks/microservices-demo/blob/main/src/checkoutservice/main.go#L120-L121) on the checkout service.
- Manual instrumentation: directly use the otel APIs to wrap your code and add instrumentation. The code you're modifying here is an example of manual instrumentation. 

We should be displaying OTEL data that customers are sending us, and that includes data collected via manual instrumentation like this. However, we shouldn't _require_ that customers need to understand how to use the OTEL APIs and understand the semantic conventions in order to light up something in the UI. Our goal with this demo environment is to reasonably reflect what we expect customers to do, which doesn't include manual instrumentation. If go instrumentation doesn't currently allow for instrumentation of http servers (it actually probably does - this is probably just old instrumentation) we shouldn't paint an overly optimistic picture by including unrealistic instrumentation in the demo environment.

Reply:
Another way to look at it is the following. In order to make this demo app more maintainable, and better match what a "typical" customer (or a customer trying out New Relic's OTel support for the first time) will see we are trying to limit the instrumentation to just the enabling of out of the box instrumentation libraries. Our assumption is that customers will only write custom code if they need to, and that they desire to keep that custom code to a minimum.

Where the reliance on instrumentation libraries break down is when you are dealing with languages like C, C++, Rust, and Go where the ability to do monkey patching isn't really a thing. That's where I think this demo app will have more "custom" instrumentation. 

With that being said, Go does have some instrumentation libraries, including one that should be able to instrument these requests, and our preference is to use that instrumentation library. If that library is not capturing the http response, we should work on getting that change made to the instrumentation library so that everyone can enjoy a better out of the box experience.